src/views/templates/details/tabs/details/components/WorkloadProfile.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+import produce from 'immer';
+import useWorkloadProfile, {
+  WorkloadProfileKeys,
+} from 'src/views/templates/list/hooks/useWorkloadProfile';
+import { ensurePath } from 'src/views/templates/utils';
+
+import { TemplateModel, V1Template } from '@kubevirt-ui/kubevirt-api/console';
+import { useModal } from '@kubevirt-utils/components/ModalProvider/ModalProvider';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { getTemplateVirtualMachineObject } from '@kubevirt-utils/resources/template';
+import { VM_WORKLOAD_ANNOTATION } from '@kubevirt-utils/resources/vm';
+import { k8sUpdate } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Button,
+  DescriptionListDescription,
+  DescriptionListGroup,
+  DescriptionListTerm,
+} from '@patternfly/react-core';
+import { PencilAltIcon } from '@patternfly/react-icons';
+
+import { TemplateDetailsGridProps } from '../TemplateDetailsPage';
+
+import WorkloadProfileModal from './WorkloadProfileModal';
+
+const WorkloadProfile: React.FC<TemplateDetailsGridProps> = ({ template, editable }) => {
+  const { createModal } = useModal();
+  const { t } = useKubevirtTranslation();
+  const workload = useWorkloadProfile(template, t);
+
+  const updateWorkload = (updatedWorkload: WorkloadProfileKeys) => {
+    const updatedTemplate = produce<V1Template>(template, (templateDraft: V1Template) => {
+      const vm = getTemplateVirtualMachineObject(templateDraft);
+      ensurePath(vm, 'spec.template.metadata.annotations');
+
+      vm.spec.template.metadata.annotations[VM_WORKLOAD_ANNOTATION] = updatedWorkload;
+
+      return templateDraft;
+    });
+
+    return k8sUpdate({
+      model: TemplateModel,
+      data: updatedTemplate,
+      ns: updatedTemplate?.metadata?.namespace,
+      name: updatedTemplate?.metadata?.name,
+    });
+  };
+
+  const onEditClick = () =>
+    createModal(({ isOpen, onClose }) => (
+      <WorkloadProfileModal
+        obj={template}
+        isOpen={isOpen}
+        onClose={onClose}
+        onSubmit={updateWorkload}
+      />
+    ));
+
+  return (
+    <DescriptionListGroup>
+      <DescriptionListTerm>{t('Workload profile')}</DescriptionListTerm>
+      <DescriptionListDescription>
+        {workload || t('Other')}
```

Comment:
```suggestion
        { t(workloadLabels[workload] || 'Other') }
```

Reply:
It is better not to use constants inside `t` function. I already discussed that within the team.

Reply:
we use constants in `t` all the time
for example `t('Other')`,
In this case you want to translate the flavor, that is prop,
you have several ways of solving this issue:
you can have a dict that return the already translated values:
``` ts
const workloadProfiles = {
  desktop: t('Desktop'),
  server:t('Server'),
  highperformance: t('High performance'),
};
```
or a dict hodding the translation keys:
```ts
// t('plugin__kubevirt-plugin~Desktop')
// t('plugin__kubevirt-plugin~Server')
// t('plugin__kubevirt-plugin~High performance')
// t('plugin__kubevirt-plugin~Other')
const workloadProfiles = {
  desktop: 'Desktop',
  server: 'Server',
  highperformance: 'High performance',
};
```
and then translate them when using them, in this case it nicer to use the tranlation keys, and translate only when rendering,
a god reason to use the translation keys, is that we get the label using a method `getWorkloadLabes` such a method is getting simpler if you don't need to do translations inside the method.
for example:
```ts
const getWorkloadLabes = (workload, t) =>
   t(workloadProfiles(workload) || 'Other');
```
vs
```ts
const getWorkloadLabes = (workload) =>
   workloadProfiles(workload) || 'Other';
``` 
a. if you use the keys instead of the translations, you do need to pass `t` arround,
b. someone calling this method may use it as a component in a bigger string and re-translate it, for example
```ts
t(`this is the workload: ${workload}`)
```

Reply:
Updated the code to achieve [your suggestion](https://github.com/kubevirt-ui/kubevirt-plugin/pull/477/files#r889840996) and without creating any extra helper like `getWorkloadLabes` as suggested just above.